File: 238-product-of-array-except-self/238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        results = [1 for i in range(len(nums))]
        
        prod =1
        for i in range(0, len(nums) -1):
            n = nums[i]
            prod *= n
            
            results[i+1] = prod
        
        prod=1
        for i in reversed(range(1, len(nums))):
            n = nums[i]
            prod *= n
            
            results[i-1] *= prod
        
        
        return results

File: 238-product-of-array-except-self/238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] res = new int[nums.length];
        
        Arrays.fill(res, 1);
        
        int prod = 1;
        
        for (int i = nums.length - 1; i>0; i-=1){
            prod *=nums[i];
            res[i-1] *= prod;
        }
        
        prod = 1;
        
        for (int i = 0; i<nums.length -1; i+=1){
            prod *=nums[i];
            res[i+1] *= prod;
        }
        
        return res;
    }
}